src/__tests__/starWarsData.ts

```diff
@@ -55,7 +55,7 @@ export function createShip(shipName: string, factionId: string): Ship {
   allShips.push(newShip);
 
   const faction = allFactions.find((obj) => obj.id === factionId);
-  faction?.ships.push(newShip.id);
+  faction.ships.push(newShip.id);
```

Comment:
this change is not needed you need to enable `strictNullChecks` in `tsconfig.json` for this to pass without changes